src/math/operations/truncateOps.ts

```diff
@@ -257,11 +259,131 @@ const ambos = makeTruncateTrio({
   },
 })
 
+function getDirection(v1: Vec3D, v2: Vec3D) {
+  return v2.sub(v1).getNormalized()
+}
+
+const augTruncate = makeOpPair({
+  graph: Composite.query
+    .where((s) => {
+      const source = s.data.source
+      return (
+        s.isAugmented() &&
+        !s.isDiminished() &&
+        source.isClassical() &&
+        source.isRegular()
+      )
+    })
+    .map((entry) => ({
+      left: entry,
+      right: entry.withData({
+        source: entry.data.source.withData({ operation: "truncate" }),
+      }),
+    })),
+  middle: "right",
+  getPose($, { geom, specs }) {
+    const source = specs.data.source
+    let caps = Cap.getAll(geom)
+    const isTetrahedron =
+      source.isClassical() && source.isTetrahedral() && source.isRegular()
+    // If source is a tetrahedron, take only the first cap (the other is the base)
+    if (isTetrahedron) {
+      caps = [caps[0]]
+    }
+    const capVertIndices = caps.flatMap((cap) =>
+      cap.innerVertices().map((v) => v.index),
+    )
+    const sourceVerts = geom.vertices.filter(
+      (v) => !capVertIndices.includes(v.index),
+    )
```

Comment:
A lot of this is repeated in the transform function, but give you a lot of information we need and I'm not sure the best way to cut it up and abstract it.